src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -70,8 +70,8 @@ public T remove(int index) {
     public boolean remove(T object) {
         Node<T> node = first;
         for (int i = 0; i < size; i++) {
-            if (node.item == object
-                    || node.item.equals(object)) {
+            if (object == null ? node.item == null
+                    : node.item.equals(object)) {
```

Comment:
Imagine:
object   ==  3; 
node == null;
what will happend?
So, you should invoke the equals method in the nonNull method. Now item can be Null and you try to invoke equals in the null object.